Update CHANGELOG.md for v.0.0.3 (#215)
informalsystems/ibc-rs: CHANGELOG.md

# Changelog

## v0.0.3

*September 1, 2020*

This release is focused on the IBC handler framework and implementation of ICS02 and ICS07.

### BREAKING CHANGES:
  
### FEATURES:

> - [modules] Proposal for IBC handler architecture and partial implementation of ICS02 and ICS03 message handlers.

Reviewer: Is ICS03 in here? I thought just 7

> - [modules] Proposal for IBC handler architecture and partial implementation of ICS02 and ICS03 message handlers.

Author: Just 7, corrected.

# Changelog

## v0.0.3

*September 1, 2020*

> This release focuses mostly on the IBC handler framework and implementation of ICS02 and ICS07.

Reviewer 2: Maybe we can start getting used to avoiding the word "handler". 
(File `adr-003-handler-implementation.md` should be renamed, btw.)

```suggestion
This release focuses mostly on the IBC message processor framework and implementation of ICS02 and ICS07.
```

# Changelog

## v0.0.3
*September 1, 2020*

This release focuses on the IBC message processor framework and initial
implementations in ICS02 and ICS07. It also introduces an initial specification for the relayer algorithm.

Other highlights:

> - A new `ibc` crate is released from the previous `modules` crate.

Reviewer 3: ```suggestion
- The modules crate is published as [ibc](https://crates.io/crates/ibc) in crates.io
```

# Changelog

## v0.0.3

*September 1, 2020*

> This release focuses on the IBC message processor framework (formerly known as handler framework) and initial implementations in ICS02 and ICS03. It also introduces an initial specification for the relayer algorithm.

Reviewer 4: Maybe we need a line break here, only I couldn't figure out where exactly is the best.

Other highlights:
- A new `ibc` crate is released from the previous `modules` crate.

> - ADR-001 is complete. 🎉

Reviewer 2: ```suggestion
- ADR-001 and ADR-003 are complete. 🎉
```

### BREAKING CHANGES:

- [modules] Renamed `modules` crate to `ibc` crate. Version number for the new crate is not reset. ([#198])
- [modules/ics02] `ConnectionId`s are now decoded to `Vec<ConnectionId>` and validated instead of `Vec<String>` ([#185])
- [modules/ics03] Removed `Connection` and `ConnectionCounterparty` traits ([#193])
- [modules/ics04] Removed `Channel` and `ChannelCounterparty` traits ([#192])

### FEATURES:
- [modules/ics02] partial implementation of message handler ([#119], [#194])
- [modules/ics03] partial implementation of message handler ([#119], [#194])

> - [architecture/ADR-003] Proposal for IBC handler architecture ([#119], [#194])

Reviewer 2: ```suggestion
- [architecture/ADR-003] Proposal for IBC handler (message processor) architecture ([#119], [#194])
```

Earlier in this changelog we called it "message processor framework". We should be consistent..

- [spec/relayer] Detailed technical specification of the relayer algorithm with focus on client update ([#84])
- [architecture/ADR-001] Documentation for the repository structure ([#1])

> - [architecture/FSM-1] Connection Handshake FSM English description added ([#122])

Reviewer 2: ```suggestion
- [architecture/FSM-1] Connection Handshake FSM English description ([#122])
```

### IMPROVEMENTS:

- [contributing] Updated CONTRIBUTING.md. Please read before opening PRs ([#195])
- [relayer-cli] Refactor ConnectionId decoding in `query client` ([#185])

> - Tests implemented for breaking changes

Reviewer 2: This begs for more context. I'm actually not sure what we mean here hmmm.

Reviewer 4: Yes, good point. I got lazy because noone from the end-users care about our tests. :) Originally, I wanted to not even include it...

For our own sake, maybe it's worth including any new tests we've done, just so we keep remembering to do tests.

Reviewer 4: ```suggestion
```

# Changelog

## v0.0.3

*September 1, 2020*

> This release focuses on the IBC message processor framework (formerly known as handler framework) and initial implementations

Author: This is the first stab at implementing this so no "formerly" :)
```suggestion
This release focuses on the IBC message processing framework and initial implementations
```

> in ICS02 and ICS03. It also introduces an initial specification for the relayer algorithm.

Author: ```suggestion
in ICS02 and ICS07. It also introduces an initial specification for the relayer algorithm.
```

Other highlights:
- A new `ibc` crate is released from the previous `modules` crate.
- ADR-001 is complete. 🎉

### BREAKING CHANGES:

- [modules] Renamed `modules` crate to `ibc` crate. Version number for the new crate is not reset. ([#198])
- [modules/ics02] `ConnectionId`s are now decoded to `Vec<ConnectionId>` and validated instead of `Vec<String>` ([#185])
- [modules/ics03] Removed `Connection` and `ConnectionCounterparty` traits ([#193])
- [modules/ics04] Removed `Channel` and `ChannelCounterparty` traits ([#192])

### FEATURES:
- [modules/ics02] partial implementation of message handler ([#119], [#194])

> - [modules/ics03] partial implementation of message handler ([#119], [#194])

Author: ```suggestion
- [modules/ics07] partial implementation of message handler ([#119], [#194])
```

Reviewer 4: Fixed

- [architecture/FSM-1] Connection Handshake FSM English description added ([#122])

### IMPROVEMENTS:

- [contributing] Updated CONTRIBUTING.md. Please read before opening PRs ([#195])
- [relayer-cli] Refactor ConnectionId decoding in `query client` ([#185])
- Tests implemented for breaking changes

### BUG FIXES:

- [modules/ics24] Identifiers limit update according to ICS specs ([#168])

[spec/relayer]: https://github.com/informalsystems/ibc-rs/blob/master/docs/spec/relayer/Relayer.md
[#84]: https://github.com/informalsystems/ibc-rs/issues/84
[architecture/ADR-001]: https://github.com/informalsystems/ibc-rs/blob/master/docs/architecture/adr-001-repo.md
[#1]: https://github.com/informalsystems/ibc-rs/issues/1
[contributing]: https://github.com/informalsystems/ibc-rs/blob/master/CONTRIBUTING.md
[#195]: https://github.com/informalsystems/ibc-rs/pull/195
[modules]: https://github.com/informalsystems/ibc-rs/tree/master/modules
[#198]: https://github.com/informalsystems/ibc-rs/issues/198
[modules/ics02]: https://github.com/informalsystems/ibc-rs/tree/master/modules/src/ics02_client
[#185]: https://github.com/informalsystems/ibc-rs/issues/185

> [modules/ics03]: https://github.com/informalsystems/ibc-rs/tree/master/modules/src/ics03_connection

Author: ```suggestion
[modules/ics07]: https://github.com/informalsystems/ibc-rs/tree/master/modules/src/ics07_tendermint
```

Reviewer 4: Good find, apparently I didn't scroll down enough in the PR...

Reviewer 4: Fixed
